test/specs/js-file.js

```diff
@@ -162,6 +162,94 @@ describe('modules/js-file', function() {
+                    'var z = "1";',
+                ].join('\n'));
+                assert(file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(file.isEnabledRule('validateQuoteMarks', 3));
+            });
+
+            it('should work with no space before jscs:ignore', function() {
+                var file = createJsFile([
+                    'var x = "1";',
+                    'var y = "1"; //jscs: ignore validateQuoteMarks',
+                    'var z = "1";',
+                ].join('\n'));
+                assert(file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(file.isEnabledRule('validateQuoteMarks', 3));
+            });
+
+            it('should work without a space before `jscs`', function() {
+                var file = createJsFile([
+                    'var x = "1";',
+                    'var y = "1"; //jscs: ignore validateQuoteMarks',
+                    'var z = "1";',
+                ].join('\n'));
+                assert(file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(file.isEnabledRule('validateQuoteMarks', 3));
+            });
+
+            it('should not re-enable rules', function() {
+                var file = createJsFile([
+                    'var a = "1";',
+                    '// jscs: disable validateQuoteMarks',
+                    'var b = "1"; // jscs: ignore validateQuoteMarks',
+                    'var c = "1";',
+                ].join('\n'));
+                assert(file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(!file.isEnabledRule('validateQuoteMarks', 3));
+                assert(!file.isEnabledRule('validateQuoteMarks', 4));
+            });
+
+            it('should also work with multiple rules', function() {
+                var file = createJsFile([
+                    '// jscs: disable validateQuoteMarks',
+                    'var a = "1"; // jscs: ignore validateQuoteMarks, anotherRule',
+                    'var b = "1";',
+                ].join('\n'));
+                assert(!file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(!file.isEnabledRule('validateQuoteMarks', 3));
+                assert(file.isEnabledRule('anotherRule', 1));
+                assert(!file.isEnabledRule('anotherRule', 2));
+                assert(file.isEnabledRule('anotherRule', 3));
+            });
+
+            it('requires a rule to be specified', function() {
+                var file = createJsFile([
+                    '// jscs: disable validateQuoteMarks',
+                    'var a = "1"; // jscs: ignore', // invalid
```

Comment:
Ah, that's not quite the problem I'm afraid!

It's easy to make a regex that will permit comments without any rule names, and this `isEnabledRule` based test will pass in that situation, the problem is with toggling on and off all rules *when* there is already specific rules enabled. This is due to there being one special rule that represents all other rules, `'*'`.

If we have 1 rule turned off, and then want to toggle on and off all rules, how do we do this? When we toggle back on all rules after the single line trailing comment we will override the previously turned off rule as there's no good way to identify it was already on, and that it needs to be disabled again on the same line we enable all rules.

This same problem happens with the existing block comments, which I think might qualify as a bug?

Either way, the intended behaviour is slightly unclear, and implementing this feature will take a sizeable refactoring of the disabled rule index system, which my colleagues and I did not feel comfortable undertaking.